obstore/python/obstore/fsspec.py

```diff
@@ -296,3 +467,50 @@ def read(self, length: int = -1) -> Any:
             data = self.fs.cat_file(self.path, self.loc, self.loc + length)
             self.loc += length
         return data
+
+
+def register(protocol: str | Iterable[str], *, asynchronous: bool = False) -> None:
```

Comment:
This can be a follow-up PR, but we should support calling `register` with no arguments, which registers for all supported backends.